controllers/v1/observationsController.js

```diff
@@ -2151,4 +2186,49 @@ module.exports = class Observations extends Abstract {
         })
     }
 
-}
\ No newline at end of file
+
+}
+
+// /**
+//   * get subEntities of matching type by recursion.
+//   * @method
+//   * @name subEntitiesWithMatchingType
+//   * @returns {Array} - Sub entities matching the type .
+//   */
+
+async function subEntitiesWithMatchingType( parentIds,entityType,matchingData ){
+    if( !parentIds.length > 0 ){
+      return matchingData;
+    }
+
+    let bodyData={
+        "parentId" : parentIds
+    };
+    
+    let entityDetails = await sunbirdService.learnerLocationSearch(bodyData);
+
+    if( !entityDetails.data.response.length > 0 && !matchingData.length > 0 ) {
```

Comment:
@VISHNUDAS-tunerlabs check the success and check "entityDetails.data" is present